libs/application-generic/src/services/resource-validator.service.ts

```diff
@@ -0,0 +1,86 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import {
+  EnvironmentRepository,
+  NotificationTemplateRepository,
+} from '@novu/dal';
+import { FeatureFlagsKeysEnum } from '@novu/shared';
+
+import { GetFeatureFlag, NotificationStep } from '../usecases';
+
+@Injectable()
+export class ResourceValidatorService {
+  private readonly MAX_STEPS_PER_WORKFLOW = 10;
+  private readonly MAX_WORKFLOWS_LIMIT = 100;
+
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private getFeatureFlag: GetFeatureFlag,
+  ) {}
+
+  async validateStepsLimit(environmentId: string, steps: NotificationStep[]) {
+    const isWorkflowLimitEnabled = await this.getFeatureFlag.execute({
+      key: FeatureFlagsKeysEnum.IS_MAX_STEPS_PER_WORKFLOW_ENABLED,
+      environmentId,
+      organizationId: 'system',
+      userId: 'system',
+    });
+
+    if (!isWorkflowLimitEnabled) {
+      return;
+    }
+
+    if (steps.length > this.MAX_STEPS_PER_WORKFLOW) {
+      throw new BadRequestException({
+        message: `Workflow steps limit exceeded. Maximum allowed steps is ${this.MAX_STEPS_PER_WORKFLOW}, but got ${steps.length} steps.`,
+        providedStepsCount: steps.length,
+        maxSteps: this.MAX_STEPS_PER_WORKFLOW,
+      });
+    }
+  }
+
+  async validateWorkflowLimit(environmentId: string) {
+    const environment = await this.getEnvironment(environmentId);
+
+    const workflowsCount = await this.notificationTemplateRepository.count({
+      _environmentId: environmentId,
+    });
+
+    if (workflowsCount < this.MAX_WORKFLOWS_LIMIT) {
+      return;
+    }
+
+    const maxWorkflowLimit = await this.getFeatureFlag.getNumber({
+      key: FeatureFlagsKeysEnum.MAX_WORKFLOW_LIMIT_NUMBER,
+      contextKey: 'environment',
+      contextId: environmentId,
+      defaultValue: this.MAX_WORKFLOWS_LIMIT,
+      context: {
```

Comment:
Why do we pass the createdAt while requesting the feature flag? Shouldn't the context be already there as part of identifying the current organization and environment to LD on each API request?

Reply:
we do not have any context that is passed to lanchdarckly, at the moment we pass only 
`kind: used/environment/organization, and its ID`